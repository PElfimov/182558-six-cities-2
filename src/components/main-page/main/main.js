import React from "react";
import Header from '../header/header';
import City from "../city/city";
import Tabs from '../../tabs-panel/tabs/tabs';

export default function Main() {
  return (
    <div className="page page--gray page--main">
      <Header />
      <main className="page__main page__main--index">
        <h1 className="visually-hidden">Cities</h1>
        <Tabs />
        <City />
      </main>
    </div>
  );
}
