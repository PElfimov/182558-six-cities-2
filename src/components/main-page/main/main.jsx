import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import Tabs from './../../tabs-panel/tabs/tabs';
import Header from './../../header/header';
import withActiveCard from './../../../hocs/with-active-card/with-active-card';
import City from './../city/city';
import {ActionCreator} from "../../../store/actions/action-creator/action-creator";

const Main = (props)=>{
  const {match, changeCity} = props;
  const name = match.params.name;
  changeCity(name);

  const WithActiveCard = withActiveCard(City);
  return (
    <div className="page page--gray page--main">
      <Header />
      <main className="page__main page__main--index">
        <h1 className="visually-hidden">Cities</h1>
        <Tabs />
        <WithActiveCard />
      </main>
    </div>
  );
};

Main.propTypes = {
  match: PropTypes.object,
  changeCity: PropTypes.func,
};

const mapDispatchToProps = (dispatch) => ({
  changeCity: (city) => {
    dispatch(ActionCreator.changeCity(city));
  }
});


export {Main};

export default connect(null, mapDispatchToProps)(Main);

