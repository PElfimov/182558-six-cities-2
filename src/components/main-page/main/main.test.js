import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {Main} from './main';
Enzyme.configure({adapter: new Adapter()});


it(`SignIn correctly renders after relaunch`, () => {
  const tree = shallow(<Main />);

  expect(toJSON(tree)).toMatchSnapshot();
});
