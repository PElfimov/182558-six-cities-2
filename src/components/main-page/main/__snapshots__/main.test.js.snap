// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn correctly renders after relaunch 1`] = `
<div
  className="page page--gray page--main"
>
  <Connect(Header) />
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <Connect(Tabs) />
    <WithActiveCard />
  </main>
</div>
`;
